avalon/nuke/pipeline.py

```diff
@@ -155,10 +155,13 @@ def process(self):
         if (self.options or {}).get("useSelection"):
             nodes = nuke.selectedNodes()
 
-        instance = [n for n in nodes
-                    if n["name"].value() in self.name] or None
+        nodes = [n for n in nodes
```

Comment:
Could this be turned into

```python
node = next((n for n in nodes if n["name"].value() in self.name), None)`
```

Also, the fallback to `nuke.allNodes()` at the top, would you _ever_ want that to accidentally happen? Seems like that could cause crazy things at some point. You might, I'm just not sure what the intended workflow is.

And, line 161 will cause error on nothing selected `None[0]`. Maybe better to report a decent error message like "Select X or Y"?

Reply:
I think this has been taken cared by some of the commits in #483, a bit hard to tell which's which, but from the [full changes](https://github.com/getavalon/core/pull/438/files#diff-c5bffa46ac90c806a167aa2ca1b96509R167-R194) could see that's been improved.

Should be shipped in next PR :P

Reply:
yes this has been fixed additionally in #438